Improve 65 - 식별 검색 조건 상세화

jgy4419 opened the pull request:
# 연관 이슈

- #65 

# 작업 내용

- 컴포넌트 내 타입 분리
- 마크검색 UI및 기능 추가
- 마크, 제형, 분할선, 제품목, 회사정보 값 기준으로 검색 잡히도록 수정
- 마크, 제형, 분할선, 제품목, 회사정보 쿼리 추가
- 마크, 제형, 분할선, 제품목, 회사정보 관련 store 로직 및 zustand로 수정
- 

# 참고 자료

-

# 비고

- 추가 리팩토링 필요.
- 마크, 알약 상세 정보 검색 결과 앱 오래 켜두면 검색 안되는 이슈. 앱 껏다 켜야됨

jgy4419 committed:
feat : 필요한 이미지 파일 및 식별검색 시 필요한 기능 추가

jgy4419 committed:
chore : recoil 파일 삭제

jgy4419 committed:
feat : 마크 데이터 가져오는 파일 추가

jgy4419 committed:
feat : 알약 검색하기 시 마크, 제형, 분할선, 정보 관련 데이터 적용되도록 추가

jgy4419 committed:
feat : 마크 모달 띄워주는 컴포넌트 추가

jgy4419 committed:
feat : 데이터 조건에 따른 컴포넌트 생성 로직 추가 (정보, 마크 데이터 등 다른 컴포넌트 적용되어야 하는 경우 실행)

jgy4419 committed:
feat : input창, button 같이 있는 컴포넌트 추가

jgy4419 committed:
feat : 모달창에서 마크 선택되면 이미지 표시해주는 컴포넌트 추가

jgy4419 committed:
feat : 알약 검섹 시 식별 검색 화면에서 선택된 값들 가져오도록 추가

jgy4419 committed:
feat : 페이지 네이션 번호 표시해주는 컴포넌트 추가

jgy4419 committed:
feat : 모달 생성 시 마크 버튼 컴포넌트 추가

jgy4419 committed:
feat : 마크 데이터 초기화 및 이미지 데이터 전역으로 관리하기 위해 store 추가

jgy4419 committed:
refactor : 모달 관련 컴포넌트 분리 작업

jgy4419 committed:
chore : recoil 관련 파일 삭제

jgy4419 committed:
type : components/atom 폴더에 있는 컴포넌트들 타입 분리

jgy4419 committed:
type : components/screens 폴더에 있는 type 분리

jgy4419 committed:
type : components/organisms 폴더에 있는 타입들 분리

jgy4419 committed:
type : components/molecules 폴더에 있는 컴포넌트 타입 분리

jgy4419 committed:
feat : 제형, 분할선, 제품명, 회사정보, 마크에 관한 쿼리 추가

jgy4419 committed:
type : components 타입 파일들 적용해놓은 거 파일 위치, 파일명 수정

jgy4419 committed:
chore : 각 컴포넌트들 타입 가져온 경로 수정

jgy4419 committed:
chore : 제형 가져오는 변수명 수정

jgy4419 committed:
chore : 마크 보여줄 때 code 값 대신 title 로 보여주도록 수정

jgy4419 committed:
feat : 마크 데이터 호출 시 검색 안됐을 때 총 3번 더 재시도 되도록 기능 추가, Toast 최상단에 안생겨서 ToastAndroid 사용

jgy4419 committed:
feat : 마크데이터, 제형 검색조건에 잡히도록 쿼리 수정

jgy4419 committed:
feat : 마크 검색 시 또는 이미지 없는 마크인 경우 이미지 없음 이미지 적용되도록 수정

jgy4419 committed:
feat : 재시도 로직 잠시 주석 처리

jgy4419 committed:
feat : 마크 선택 시 제거도 할 수 있도록 x버튼 추가

YongSiHun commented on src/api/db/query.ts:
```diff
@@ -64,6 +73,111 @@ function getQueryForSearch(param: TPillSearchParam | null) {
+    if (markFilters.length > 0) {
+      filters.push('(' + markFilters.join(' OR ') + ')');
+    }
+  }
```
추가로 각 조건 별 SQL의 어떤 문을 의도한건지 주석으로 각각 쓰면 
차후 리팩토링 시 혼동이 없을 것 같습니다

YongSiHun committed:
feat: 데이터베이스 쿼리 개선

- 데이터베이스 검색 필터 생성 로직 가독성 개선 및 쿼리 통일
- LIKE '%데이터%' 형식으로만 검색하도록 개선
- 회사 정보 > 제조사명 으로 용어 변경

#65

jgy4419 committed:
chore : package.json에 version 명시된 거 제거, mark.ts에서 type import 가져오는 방식 수정, 마크 데이터 불러오는 함수 주석 수정
